Solve (4-6). PHC - Leetcode 3, 202, 101

gabliw opened the pull request:
Leet Code 3: Longest Substring

https://leetcode.com/problems/longest-substring-without-repeating-characters/

Runtime 558 ms
Beats 11.51%

Memory 39.7 MB
Beats 12.13%

Leetcode 101: Symmetric Tree

https://leetcode.com/problems/symmetric-tree/

Runtime 43 ms
Beats 38.42%

Memory 16.5 MB
Beats 9.95%

Leetcode 202: Happy Number

https://leetcode.com/problems/happy-number/

Runtime 41 ms
Beats 48.8%

Memory 16.5 MB
Beats 5.96%

gabliw committed:
2023.05.20 Longest Substring

Runtime 558 ms
Beats 11.51%

Memory 39.7 MB
Beats 12.13%

https://leetcode.com/problems/longest-substring-without-repeating-characters/

gabliw committed:
2023-05-20
Leetcode 101: Symmetric Tree
Park Hee Chan

Runtime 43 ms
Beats 38.42%

Memory 16.5 MB
Beats 9.95%

https://leetcode.com/problems/symmetric-tree/

gabliw committed:
2023-05-20
Leetcode 102: Happy Number
Park Hee Chan

Runtime 41 ms
Beats 48.8%

Memory 16.5 MB
Beats 5.96%

https://leetcode.com/problems/happy-number/

gabliw committed:
2023-05-20
Leetcode 102: Happy Number
Park Hee Chan

Runtime 41 ms
Beats 48.8%

Memory 16.5 MB
Beats 5.96%

https://leetcode.com/problems/happy-number/

phc-alchera commented:
Leetcode 202: Happy Number 처럼 풀었는데
생각한 것보다 동작이 늦은 것 같습니다.
제가 한 방식이 틀린건지 다른 분들 의견 부탁드립니다.

SeungHwa-Alchera commented on leetcode_3/phc.py:
```diff
@@ -0,0 +1,44 @@
+        return 1
+
+    point = 1
+    for c1 in s:
+        substring = c1
+        memoization.append(len(c1))
+        for c2 in s[point:]:
+            if c2 in substring:
+                break
+
+            print(substring, memoization)
+            substring += c2
+            memoization.append(len(substring))
```
memoization으로 길이를 비교하는것도 좋은 방법이지만 문자를 캐싱한다면 더 빠르게 풀 수 있지 않을까 싶습니다!!

SeungHwa-Alchera commented on leetcode_202/phc.py:
```diff
@@ -0,0 +1,42 @@
+        if sum_of_squares == 1:
+            return True
+
+        return self.happy_cycle(sum_of_squares, memoization + [n])
```
메모이제션을 Dictionary { num: num**2  } 형식으로 쓰면 조금 더 빨라지지 않을까요?

cys-alchera commented:
C++로 안 풀어서 그렇습니다. :smile:

gabliw committed:
2023-05-27
LeeetCode 203. Remove Linked List Elements
Park Hee Chan
https://leetcode.com/problems/remove-linked-list-elements/

Runtime 77 ms
Beats 52.19%

Memory 19.9 MB
Beats 44.22%

--

206. Reverse Linked List
Park Hee Chan
https://leetcode.com/problems/reverse-linked-list/

Runtime 53 ms
Beats 28.14%

Memory 17.9 MB
Beats 26.81%

--

876. Middle of the Linked List
Park Hee Chan
https://leetcode.com/problems/middle-of-the-linked-list/

Runtime 48 ms
Beats 22.10%

Memory 16.3 MB
Beats 8.67%
